client/vaultclient/vaultclient.go

```diff
@@ -382,42 +297,39 @@ func (c *vaultClient) Stop() {
 // The return value is a map containing all the unwrapped tokens indexed by the
 // task name.
 func (c *vaultClient) DeriveToken(alloc *structs.Allocation, taskNames []string) (map[string]string, error) {
-	var result *multierror.Error
-
 	if !c.running {
-		result = multierror.Append(fmt.Errorf("vault client is not running"))
-		return nil, result
+		return nil, fmt.Errorf("vault client is not running")
 	}
 
 	if alloc == nil {
-		result = multierror.Append(fmt.Errorf("nil allocation"))
-		return nil, result
+		return nil, fmt.Errorf("nil allocation")
 	}
+
 	if taskNames == nil || len(taskNames) == 0 {
-		result = multierror.Append(fmt.Errorf("missing task names"))
-		return nil, result
+		return nil, fmt.Errorf("missing task names")
 	}
 
-	found := false
 	verifiedTasks := []string{}
-	// Check if the given task names actually exist in the allocation
-	for _, taskName := range taskNames {
-		found = false
-		for _, group := range alloc.Job.TaskGroups {
-			for _, task := range group.Tasks {
-				if task.Name == taskName {
-					found = true
+	found := false
+	// Check if the given task names actually exist in the allocation under
+	// the correct group name
+	for _, group := range alloc.Job.TaskGroups {
```

Comment:
Use this: https://github.com/hashicorp/nomad/blob/master/nomad/structs/structs.go#L1246

Sorry should have linked it from the get-go, forgot that the task groups wasn't a map